src/consensus/dkg.rs

```diff
@@ -368,6 +357,117 @@ struct Participant {
     timer_token: u64,
 }
 
+impl Participant {
+    fn process_message(
+        &mut self,
+        node: &Node,
+        dkg_key: DkgKey,
+        message: DkgMessage,
+    ) -> Result<Vec<Command>> {
+        if self.dkg_key != dkg_key {
+            return Ok(vec![]);
+        }
+
+        let responses = self
+            .key_gen
+            .handle_message(&mut rand::thread_rng(), message)
+            .unwrap_or_default();
+
+        let mut commands = vec![];
+        for response in responses {
+            commands.extend(self.broadcast(node, dkg_key, response)?);
+        }
+
+        Ok(commands)
+    }
+
+    fn broadcast(
+        &mut self,
+        node: &Node,
+        dkg_key: DkgKey,
+        dkg_message: DkgMessage,
+    ) -> Result<Vec<Command>> {
+        let recipients: Vec<_> = self
+            .elders_info
+            .as_ref()
+            .into_iter()
+            .flat_map(EldersInfo::peers)
+            .filter(|peer| *peer.name() != node.name())
+            .map(Peer::addr)
+            .copied()
+            .collect();
+
+        trace!(
+            "broadcasting DKG message {:?} to {:?}",
+            dkg_message,
+            recipients
+        );
+
+        let variant = Variant::DKGMessage {
+            dkg_key,
+            message: bincode::serialize(&dkg_message)?.into(),
+        };
+        let message = Message::single_src(node, DstLocation::Direct, variant, None, None)?;
+
+        let mut commands = vec![];
+        commands.push(Command::send_message_to_targets(
+            &recipients,
+            recipients.len(),
+            message.to_bytes(),
+        ));
+        commands.extend(self.process_message(node, dkg_key, dkg_message)?);
+
+        Ok(commands)
+    }
+
+    fn check(&mut self) -> Option<Command> {
+        let _ = self.elders_info.as_ref()?;
```

Comment:
better at least have a log for None case ?

Reply:
That would happen when the DKG already completed but we are still handling some messages because some of the other participants have not completed yet. Putting a log there might cause log spam and I don't think it's that useful anyway. Or do you think it is?

Reply:
I am mainly agaisnt having an empty `?` , especially it is needed below.
So when this merged with the one below, the log itself is not necessary